specs/docs/security/DataHubSecurity.md

```diff
@@ -0,0 +1,267 @@
+---
+tags: [security]
+---
+
+# Data Hub Security
+
+This document briefly covers best practices to follow in MarkLogic security with links to more indepth documentation. In addition, this document covers the privileges given to each role and the rationale behind the assigned privilege.
+
+## MarkLogic Security Practices
+
+### Checking Execute Privileges
+
+Execute privileges are used to protect functions from being called by users without the proper level of authorization. Functions can be guarded by calling the [`xdmp.securityAssert`](https://docs.marklogic.com/xdmp.securityAssert) function at the beginning of its execution to ensure the user has the proper execution privileges. See [Protecting Your XQuery and JavaScript Code with Execute Privileges](https://docs.marklogic.com/guide/security/execute#id_69547).
+
+#### Example xdmp.securityAssert call
+```javascript
+xdmp.securityAssert("http://marklogic.com/data-hub/function-access", "execute");
```

Comment:
I'm all for guarding every DS endpoint with this. On one project, we wrote a little tool to parse the xdmp.securityAssert call on the first line that it was found in an endpoint module and then constructed an HTML table output of every endpoint + the privilege that it required. Very handy, may be worth doing in the future.